module/exporter-shopware-6/src/Infrastructure/Mapper/ProductCrossSelling/ProductCrossSellingChildrenMapper.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/*
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\ExporterShopware6\Infrastructure\Mapper\ProductCrossSelling;
+
+use Ergonode\Core\Domain\ValueObject\Language;
+use Ergonode\Exporter\Domain\Entity\Export;
+use Ergonode\ExporterShopware6\Domain\Entity\Shopware6Channel;
+use Ergonode\ExporterShopware6\Domain\Repository\Shopware6ProductRepositoryInterface;
+use Ergonode\ExporterShopware6\Infrastructure\Mapper\ProductCrossSellingMapperInterface;
+use Ergonode\ExporterShopware6\Infrastructure\Model\AbstractProductCrossSelling;
+use Ergonode\ExporterShopware6\Infrastructure\Model\Basic\AssignedProduct;
+use Ergonode\ProductCollection\Domain\Entity\ProductCollection;
+use Ergonode\ProductCollection\Domain\Entity\ProductCollectionElement;
+
+class ProductCrossSellingChildrenMapper implements ProductCrossSellingMapperInterface
+{
+    private Shopware6ProductRepositoryInterface $shopware6ProductRepository;
+
+    public function __construct(Shopware6ProductRepositoryInterface $shopware6ProductRepository)
+    {
+        $this->shopware6ProductRepository = $shopware6ProductRepository;
+    }
+
+    public function map(
+        Shopware6Channel $channel,
+        Export $export,
+        AbstractProductCrossSelling $shopware6ProductCrossSelling,
+        ProductCollection $productCollection,
+        ProductCollectionElement $collectionElement,
+        ?Language $language = null
+    ): AbstractProductCrossSelling {
+
+        foreach ($productCollection->getElements() as $element) {
+            if ($element === $collectionElement) {
+                continue;
+            }
+
+            $shopware6ProductCrossSelling = $this->mapElement(
+                $channel,
+                $shopware6ProductCrossSelling,
+                $element,
+                $this->getCurrentPosition($shopware6ProductCrossSelling)
+            );
+        }
+
+        return $shopware6ProductCrossSelling;
+    }
+
+    private function mapElement(
+        Shopware6Channel $channel,
+        AbstractProductCrossSelling $shopware6ProductCrossSelling,
+        ProductCollectionElement $collectionElement,
+        int $position = 1
+    ): AbstractProductCrossSelling {
+        $shopwareId = $this->shopware6ProductRepository->load($channel->getId(), $collectionElement->getProductId());
```

Comment:
`load` method returns just id?

Reply:
Yes, In this time

Reply:
I'd suggest to rename it appropriately not to be confused. It finds id rather, not loads